### PR TITLE
fix bug about consensus

### PR DESCRIPTION
**Pull request by mask-pp**

### CODE REVIEW

**Comment by ChuhanJin** [comment]
Can one of the admins verify this patch?

File: miner/worker.go [hunk 1]
```diff
@@ -400,7 +400,7 @@ func (w *worker) newWorkLoop(recommit time.Duration) {
 		case head := <-w.chainHeadCh:
 			clearPending(head.Block.NumberU64())
 			timestamp = time.Now().Unix()
-			// commit(false, commitInterruptNewHead)
+			commit(true, commitInterruptNewHead)
```
Comment (0xmountaintop) [inline miner/worker.go:403]: any quick way to understand the effect of `true`/`false` here?

originally it's `false`, right? why we want to change it to `true` here

**Review by jules — COMMENTED** [review]
LGTM, pending an answer to haoyu's comment

**Comment by mask-pp** [comment]
> Can one of the admins verify this patch?

File: miner/worker.go (hunk 1 above)
Comment (mask-pp) [reply miner/worker.go:403]: This involves multiple changes, and we need to use transaction-driven block generation, where true means no new blocks are generated when there are no transactions in the pool.
The problem with the previous change is that when the volume exceeds the upper limit, new blocks are no longer generated.
The easiest way to do this is based on the Ethereum source code by simply changing 'commit(false,' to true.

**Comment by mask-pp** [comment]
> Can one of the admins verify this patch?

You can spot the problem when you run a pressure test on a large number of txs.

File: miner/worker.go (hunk 1 above)
Comment (mask-pp) [reply miner/worker.go:403]: > any quick way to understand the effect of `true`/`false` here?
> 
> originally it's `false`, right? why we want to change it to `true` here

Line403 is the key change.

Comment (0xmountaintop) [reply miner/worker.go:403]: I see, so basically the controlling "no new blocks are generated when there are no transactions in the pool" logic is here: https://github.com/scroll-tech/go-ethereum/blob/e9a5a982d65077d2bd4ad59bf67ed53c6b285ee1/miner/worker.go#L1012-L1016

**Review by 0xmountaintop — APPROVED** [review]

File: miner/worker.go (hunk 1 above)
Comment (mask-pp) [reply miner/worker.go:403]: Yes, that right.

**Pull request merged**